homeassistant/components/rympro/__init__.py

```diff
@@ -32,12 +31,12 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     except UnauthorizedError:
         try:
             token = await rympro.login(data[CONF_EMAIL], data[CONF_PASSWORD], "ha")
-            hass.config_entries.async_update_entry(
-                entry,
-                data={**data, CONF_TOKEN: token},
-            )
         except UnauthorizedError as error:
             raise ConfigEntryAuthFailed from error
+        hass.config_entries.async_update_entry(
+            entry,
+            data={**data, CONF_TOKEN: token},
+        )
 
     coordinator = RymProDataUpdateCoordinator(hass, rympro, SCAN_INTERVAL)
```

Comment:
If the scan interval is a constant, we can use the constant directly where needed.

Reply:
I initially had it here as `60*60`, but was asked to change it in the review to the previous PR.

Reply:
I mean we don't need a parameter here. Just pass the constant to the coordinator base class.